lib/shmem.c

```diff
@@ -38,10 +38,10 @@ int metal_shmem_open_generic(const char *name, size_t size,
 		shmem = metal_container_of(node, struct metal_generic_shmem, node);
 		if (strcmp(shmem->name, name) != 0)
 			continue;
-		if (size > metal_io_region_size(&shmem->io))
-			continue;
-		*result = &shmem->io;
-		return 0;
+		if (size < metal_io_region_size(&shmem->io)) {
```

Comment:
What about the case where size==metal_io_region_size()? This change makes the code behave differently. Maybe "<="?

Reply:
fixed, thanks!